src/envoy/http/mixer/filter.cc

```diff
@@ -30,6 +31,49 @@ namespace Envoy {
 namespace Http {
 namespace Mixer {
 
+namespace {
+
+// detect gRPC message boundaries: each message is prefixed by 5 bytes
+// length-prefix https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
+void incrementCounter(Buffer::Instance& data, GrpcMessageCounter* counter) {
```

Comment:
definitely need a test for this method

Reply:
OK. If it helps, there is an integration test https://github.com/istio/istio/pull/12195. There is definite lack of any unit testing in this particular part of the code, but we can add something for this specific function.

Reply:
Added a unit test.